### ISSUE TITLE
"Unsupported Media Type" error when exporting all-day events through calcurse-caldav

### ISSUE THREAD

**Issue by maxigaz** [lfos/calcurse#94]
When I create all-day events in calcurse and try to sync them with a Nextcloud calendar using calcurse-caldav, I get the following error message:

```
Connecting to cloud.disroot.org...
Loading synchronization database from /home/xxx/.calcurse/caldav/sync.db...
Pushing new object d2c91457a872abffe7935a222537fc430051f190 to the server.
error: The server at cloud.disroot.org replied with HTTP status code 415
error: (Unsupported Media Type) while trying to access https://cloud.disroot.org
error: /remote.php/dav/calendars/<username>/<calendar-id>
error: /d2c91457a872abffe7935a222537fc430051f190.ics.
```

If I delete those events newly created in calcurse, replace them with events with specific start and end times, and run calcurse-caldav again, it completes without errors.

Another thing I noticed is that there's no issue when all-day events created in the Nextcloud web interface or Android's calendar are imported into calcurse. They just appear with an asterix in front of them in the list:

```
* Example event created outside calcurse
  Example event created in calcurse
```

I don't know if it's related to #46, so I'm creating a separate issue for now.

Using calcurse v4.3.0 under Manjaro Linux.

**Comment by lfos** [comment]
It looks like the first two dates originate from all-day events and the others from appointments (with a starting time). Isn't that the case?

**Comment by hiqua** [comment]
@loweryaustin can you try by using the format I mentioned in #81, by adding `;VALUE=DATE`:

```
       DTSTART;VALUE=DATE:20070628
       DTEND;VALUE=DATE:20070709
```

Just to be sure that solving #81 would actually solve your issue with nextcloud (that I do not use personally).

**Comment by loweryaustin** [comment]
Sure thing, I have to wait till I get home to try it. Probably should hear back from me around 16:00 CST.

**Comment by loweryaustin** [comment]
@hiqua I tried exporting the event with: `calcurse -x ical --export-uid --filter-hash=28013ef7148261da9f566d617edfd648f7920592 > test.ical`
I then tried importing it through the NextCloud GUI interface. No matter what I did to try to import it I ran into a different validation error that is unrelated to this issue, so I don't really have a good way to test your change.

**Comment by hiqua** [comment]
Sorry my request was confusing. What I meant is: can you try to add ';VALUE=DATE' manually in your `test.ical` file, by editing the file?

To obtain:
```
 DTSTART;VALUE=DATE:20180501
 DTSTART;VALUE=DATE:20180503
```
as the two first lines, instead of:
```
DTSTART:20180501
DTSTART:20180503
```
(do the same for DTEND)

If you can then import the file normally in NextCloud, it means that this issue and #81 are the same.

**Comment by hiqua** [comment]
@loweryaustin you can also try directly with this version: https://github.com/lfos/calcurse/pull/97

Does it solve the issue?

**Comment by aakvik** [comment]
Just tested pu branch and #97 has solved the problem for me in owncloud/nextcloud.

**Comment by lfos** [comment]
Fixed in 0114289 (ical: use the VALUE parameter in DTSTART, 2018-05-09).
